network/records/subnets.go

```diff
@@ -105,7 +105,14 @@ func SharedSubnets(a, b []byte, maxLen int) []int {
 	if len(a) == 0 || len(b) == 0 {
 		return shared
 	}
```

Comment:
Perhaps a bit clearer would be to `return nil` here, and define `var shared []int` down below (closer to the actual place of usage).

Reply:
@iurii-ssv I agree, thanks, but we've decided not to touch this function in this PR and find the root cause of the issue. However, we are going to refactor it in another PR

Reply:
@iurii-ssv refactored in https://github.com/ssvlabs/ssv/pull/1813, please review it